src/main/java/com/redhat/cloud/policies/app/rest/PolicyCrudService.java

```diff
@@ -512,27 +531,41 @@ public Response deletePolicies(List<UUID> uuids) {
         // TODO Temp counter used to investigate the engine instability, remove ASAP.
         registry.counter("policies_ui_deletePolicies", new Tag("account", user.getAccount())).inc();
 
+        // TODO POL-649 Do we really need to return this?
         List<UUID> deleted = new ArrayList<>(uuids.size());
 
-        for (UUID uuid : uuids) {
-            Policy policy = Policy.findById(user.getAccount(), uuid);
-            if (policy == null) {
-                // Nothing to do for us
+        if (lightweightEngineConfig.isEnabled()) {
+            Set<UUID> reallyDeleted = new HashSet<>(uuids.size());
+            for (UUID uuid : uuids) {
+                Policy policy = Policy.findById(user.getAccount(), uuid);
                 deleted.add(uuid);
-            } else {
-                boolean deletedOnEngine = false;
-                try {
-                    engine.deleteTrigger(policy.id, user.getAccount());
-                    deletedOnEngine = true;
-                } catch (NotFoundException nfe) {
-                    // Engine does not have it - we can delete anyway
-                    deletedOnEngine = true;
-                } catch (Exception e) {
-                    log.warn("Deletion on engine failed", e);
-                }
-                if (deletedOnEngine) {
+                if (policy != null) {
                     policy.delete();
+                    reallyDeleted.add(uuid);
```

Comment:
are you sure it was really really really deleted? :grin:

Reply:
I learned from the Camel integration in notifications 😄